typescript/src/services/convertPrivateClassesToInterfaces.ts

```diff
@@ -1,22 +1,27 @@
 import { DocCollection } from 'dgeni';
+import { ClassExportDoc } from '../api-doc-types/ClassExportDoc';
 import { ConstExportDoc } from '../api-doc-types/ConstExportDoc';
+
 export function convertPrivateClassesToInterfaces(exportDocs: DocCollection, addInjectableReference: boolean) {
 
   exportDocs.forEach(exportDoc => {
 
     // Search for classes with a constructor marked as `@internal`
     if (exportDoc.docType === 'class' && exportDoc.constructorDoc && exportDoc.constructorDoc.internal) {
+      const classDoc = exportDoc as ClassExportDoc;
```

Comment:
We should rather use a type constraint function here:

```
function isPrivateClassExportDoc(doc as any): doc is ClassExportDoc {
  return exportDoc.docType === 'class' && exportDoc.constructorDoc && exportDoc.constructorDoc.internal;
}
```

This would be called as:

```
if (isPrivateClassExportDoc(exportDoc)) {
  ...
}
```
Then inside the if statement the `exportDoc` will already be typed as `ClassExportDoc`

Reply:
Agreed. Done!